### PR TITLE
fix(ZNTA-931): incorrect scopes

### PR DESCRIPTION
**Pull request by broccolinisoup**
I am inspired of TODO statement which was "// need to get username from DAO due to lazy loading of account in key". I got the username from AccountDAO instead of through HAccountResetPasswordKey entity.

### CODE REVIEW

**Comment by djansen-redhat** [comment]
Did this get based off an incorrect branch?

**Comment by carlosmunoz** [comment]
@djansen-redhat Doesn't look like it... what makes you think that?

**Comment by broccolinisoup** [comment]
@djansen-redhat Yea at first I accidentally created it from ZNTA-470 but I revert it later and in this morning I caught up the origin master. @carlosmunoz  maybe from commits?

**Comment by djansen-redhat** [comment]
@carlosmunoz I'm seeing commits for both contact admin fix (not yet merged) and reset password fix

**Comment by broccolinisoup** [comment]
@carlosmunoz Could you review it? I cannot also add a label.

**Comment by carlosmunoz** [comment]
So, I'm actually tempted to say that this was not necessary at all after looking at the code, and for the following reasons:

1. The purpose of the comment in the code was not to flag a TODO, but rather to explain why there is a need to query the data from the database instead of getting it from the `HAccountResetPasswordKey` object already in memory. The code is already performing such a query to fetch the `HAccount`'s details.

2. The query being performed in the current code is actually more efficient than the proposed one in the pull request; It is a 2-table join with a primary key filter. The proposed one is doing a correlated subquery (a query in the where clause) which is not nearly as efficient (although it has its uses).

**Comment by seanf** [comment]
@carlosmunoz makes good points, but for next time, it's always good to give the pull request a meaningful name, eg "Znta 931" might be better as "ZNTA-931 get username from DAO".

@broccolinisoup also, please have a look at this guide to good commit messages (use the imperative!): http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

And https://github.com/zanata/zanata-server/blob/master/CONTRIBUTING.md#-commit-message-format

**Comment by broccolinisoup** [comment]
@seanf Alex sent me these links a few days ago which I sent a commit with this format : ( fix(ZNTA-960): modify try catch block) I will follow both commit message and pull req format. Thank you.

@carlosmunoz I completely understood you point. Which one is the query being performed that you mentioned? I couldn't find a proper function -apparently I missed it - so that is the reason why I wrote a new one.

File: zanata-war/src/main/java/org/zanata/action/PasswordResetAction.java [hunk 1]
```diff
@@ -108,9 +110,9 @@ public void validateActivationKey() {
 
     @Transactional
     public String changePassword() {
-        // need to get username from DAO due to lazy loading of account in key
         String username =
-            accountResetPasswordKeyDAO.getUsername(key.getKeyHash());
```
Comment (carlosmunoz) [inline zanata-war/src/main/java/org/zanata/action/PasswordResetAction.java]: @broccolinisoup This is the query I was referring to. If you follow the `getUsername` method on this DAO you'll see the query.

**Comment by broccolinisoup** [comment]
@carlosmunoz I just committed new changes that we did yesterday together. Could you take a quick look again?

**Comment by carlosmunoz** [comment]
:+1: 
(let's see what Jenkins thinks)

**Comment by broccolinisoup** [comment]
Seems that jenkins liked it.  The next step is testing? @djansen-redhat @carlosmunoz

**Comment by broccolinisoup** [comment]
Move to https://github.com/zanata/zanata-server/pull/1159

**Pull request closed without merging**